src/drop.test-d.ts

```diff
@@ -0,0 +1,54 @@
+import { drop } from "./drop";
+import { pipe } from "./pipe";
+
+describe("data first", () => {
+  it("works on regular inputs", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 2)).toEqualTypeOf<[3, 4, 5]>();
+  });
+
+  it("works on empty arrays", () => {
+    expectTypeOf(drop([], 2)).toEqualTypeOf<[]>();
+  });
+
+  it("works with negative numbers", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], -1)).toEqualTypeOf<[5]>();
+  });
+
+  it("works when dropping more than the length of the array", () => {
+    expectTypeOf(drop([1, 2, 3, 4, 5], 10)).toEqualTypeOf<[]>();
+  });
+
+  it("works with generalized typed arrays", () => {
+    expectTypeOf(drop([] as Array<number>, 2)).toEqualTypeOf<[]>();
```

Comment:
this is incorrect too, the return type for dropping from a generic array should be that array, and not an empty array...

Reply:
Fixed